packages/vuetify/src/composables/rounded.ts

```diff
@@ -1,35 +1,41 @@
 // Utilities
-import { computed } from 'vue'
+import { computed, isRef } from 'vue'
 import { propsFactory } from '@/util'
 
 // Types
+import type { Ref } from 'vue'
+
+type RoundedValue = boolean | string | number | null | undefined
+
 export interface RoundedProps {
-  rounded?: boolean | string | number | null
+  rounded?: RoundedValue
   tile?: boolean
 }
 
+type RoundedData = {
+  roundedClasses: Ref<string[]>
+}
+
 // Composables
 export const makeRoundedProps = propsFactory({
   rounded: {
     type: [Boolean, Number, String],
     default: undefined,
   },
-  tile: Boolean,
 }, 'rounded')
 
-export function useRounded (props: RoundedProps, name: string) {
+export function useRounded (props: RoundedProps | Ref<RoundedValue>, name: string): RoundedData {
   const roundedClasses = computed(() => {
+    const rounded = isRef(props) ? props.value : props.rounded
```

Comment:
Same thing here as above. This logic doesn't belong in these types of components.